src/usermod.c

```diff
@@ -751,6 +755,14 @@ static void update_group (void)
 			continue;
 		}
 
+		/*
+		* If rflg+Gflg  is passed in AKA -rG invert is_member flag, which removes
+		* mentioned groups while leaving the others.
+		*/
+		if (Gflg && rflg && ((!is_member && was_member) || (is_member && was_member))) {
```

Comment:
Sorry - I thin you fixed this below, but not here:  

((!is_member && was_member) || (is_member && was_member))

should just be the same as (was_member)

Reply:
Ahh, apologies, I missed it.